dali/base/dadfs.cpp

```diff
@@ -7929,6 +7930,130 @@ IDistributedSuperFile *CDistributedFileDirectory::createSuperFile(const char *_l
+    {
+        Owned<IPropertyTreeIterator> treeIter = tree->getElements("SubFile");
+        ForEach(*treeIter)
+        {
+            IPropertyTree & st = treeIter->get();
+            StringBuffer subfilename;
+            st.getProp("@name", subfilename);
+            CDfsLogicalFileName sublname;
+            sublname.set(subfilename.str());
+            if (!sublname.isExternal() && !sublname.isForeign())
+            {
+                StringBuffer subquery;
+                sublname.makeFullnameQuery(subquery, DXB_File, true);
+                Owned<IRemoteConnection> subconn = querySDS().connect(subquery.str(), myProcessSession(), RTM_LOCK_WRITE, FOREIGN_DALI_TIMEOUT);
+                if (!subconn)
+                {
+                    sublname.makeFullnameQuery(subquery.clear(), DXB_SuperFile, true);
+                    subconn.setown(querySDS().connect(subquery.str(), myProcessSession(), 0, FOREIGN_DALI_TIMEOUT));
+                }
+                if (!subconn)
+                {
+                    ERRLOG("createSuperFile: SuperFile '%s' is missing SubFile file '%s'", pSuperName, subfilename.str());
+                    continue;
+                }
+                subnum++;
+
+                Owned<IPropertyTree> subroot = subconn->getRoot();
+                Owned<IPropertyTreeIterator> iter = subroot->getElements("SuperOwner");
+                StringBuffer pname;
+                bool parentok=false;
+                ForEach(*iter)
+                {
+                    iter->query().getProp("@name", pname.clear());
+                    if (strcmp(pname.str(), pSuperName) == 0)
+                        parentok = true;
+                    else
+                    {
+                        CDfsLogicalFileName sdlname;
+                        sdlname.set(pname.str());
+                        StringBuffer sdquery;
+                        sdlname.makeFullnameQuery(sdquery, DXB_SuperFile, true);
+                        Owned<IRemoteConnection> sdconn = querySDS().connect(sdquery.str(), myProcessSession(), 0, FOREIGN_DALI_TIMEOUT);
+                        if (!conn)
+                            WARNLOG("createSuperFile: SubFile '%s' has missing owner SuperFile '%s'", sublname.get(), sdlname.get());
+                    }
+                }
+                if (!parentok)
+                {
+                    WARNLOG("createSuperFile: SubFile '%s' is missing link to SuperFile '%s'", sublname.get(), pSuperName);
+                    Owned<IPropertyTree> subFileTree = createPTree("SuperOwner");
+                    subFileTree->setProp("@name", pSuperName);
+                    subroot->addPropTree("SuperOwner", subFileTree.getClear());
+
+                }
+                subconn->commit();
+                subconn->close();
+            }
+        }
+    }
+    PROGLOG("%d of %d SubFile(s) reference to SuperFile '%s' are fixed.", subnum, numSubfiles, pSuperName);
+
+    return superfile.getLink();
```

Comment:
I can't see where 'superfile' is set anywhere.. ?
See previous comment, should return a valid but unattached IDistributeSuperFile, which can then be attach()'d